soc/arm/ambiq/apollo4x/Kconfig.series

```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+
+# Ambiq Apollo4 MCU Series
+
+config SOC_SERIES_APOLLO4X
+	bool "Apollo4 Series MCU"
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select CPU_HAS_FPU
+	select SOC_FAMILY_AMBIQ
+	select HAS_SWO
+	select AMBIQ_HAL
```

Comment:
this needs to be selected by drivers if needed. see e.g. st/gd

Reply:
I've modified HAL Kconfig so that specific HAL drivers can be selected independently by Zephyr drivers:
https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/drivers/timer/Kconfig.ambiq#L11
https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/drivers/pinctrl/Kconfig.ambiq#L9

But the SoC init also calls Ambiq HAL functions, so I think the main HAL config needs to be selected here?